### PR TITLE
Add troubleshooting page for unsupported operations in Mountpoint

### PR DESCRIPTION
**Pull request by sauraank**
## Description of change
Created a troubleshooting page in doc section for unsupported operations in Mountpoint.

Made the list of Unsupported Operations and its error code. Linked the page in README.md. Added its link in bug-report template along with instruction to provide logs.
Opened this PR as we have made the release.
NOTE - I have added the changes in PR #718 as unreleased section.

Relevant issues: 
#533 
## Does this change impact existing behavior?
No.

This is not a breaking change.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -50,6 +50,10 @@ body:
       required: true      
   - type: textarea
     id: behavior
+    - type: markdown
+    attributes:
+      value: |
+        Please check the list of unsupported Operations and their errors from the [Troubleshooting page](doc/TROUBLESHOOTING.md) .
```
Comment (sauraank) [inline .github/ISSUE_TEMPLATE/bug-report.yml:56]: Not sure if adding like this in yaml file works.

File: doc/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Mountpoint only supports sequential writes. On random out of order writes users should expect `Invalid Argument`. For example - 
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+In the logs, we get the following WARN message - 
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed:\
+upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:16]: Let's format the markdown with more spacing around headers and code blocks.

Also, we should prefer passive voice in the documentation where possible.

````suggestion
## Random writes

Mountpoint supports writing to a file sequentially. Random writes, or 'out-of-order' writes, will return an error to FUSE, which may appear in applications with the error message "Invalid argument".

For example, the following code seeks one byte into the file and writes a single byte using `dd`.

```
$ dd if=/dev/random of=out seek=1 count=1
dd: writing to 'out': Invalid argument
```

In Mountpoint's logs, a warning message will be emitted similar to below:

```
WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed: upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
```
````

File: doc/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Writing to an existing File 
+Mountpoint does support overwriting to existing file using mount option `--allow-overwrite` . Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+In the logs, we get the following WARN message - 
+```
+WARN file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:23]: I liked the example you gave above with `dd`. Can you have one with something like...:

```sh
echo "Overwriting a file..." > /mnt/s3-bucket/existing-file.txt
```

nit: "Mountpoint supports" instead of "Mountpoint does support"

File: doc/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## Deleting file (`rm` without mount option `--allow-delete`)
+In order to delete files using Mountpoint user need to enable it using `--allow-delete` mount option. To know more about the behaviour of file deletion, please visit [Delete Semantics](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes)
+Without the flag, if a customer tries to run  `rm test-file.txt` command, they will see the following error: `rm: cannot remove 'test-file.txt': Operation not permitted`
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:27]: Show a full example as a separate block.

File: doc/TROUBLESHOOTING.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+## Writing to an existing File 
+Mountpoint does support overwriting to existing file using mount option `--allow-overwrite` . Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+In the logs, we get the following WARN message - 
+```
+WARN file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
+
+## Deleting file (`rm` without mount option `--allow-delete`)
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:25]: ditch the bit in brackets

File: doc/TROUBLESHOOTING.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+In order to delete files using Mountpoint user need to enable it using `--allow-delete` mount option. To know more about the behaviour of file deletion, please visit [Delete Semantics](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes)
+Without the flag, if a customer tries to run  `rm test-file.txt` command, they will see the following error: `rm: cannot remove 'test-file.txt': Operation not permitted`
+In the logs, this will show up as 
+```
+WARN unlink{req=8 parent=1 name="test-file.txt"}: mountpoint_s3::fuse: unlink failed: Deletes are disabled. \
+Use '--allow-delete' mount option to enable it.
+```
+
+## Listing file which is shadowed by directory of same name
+A file system directory cannot contain both a file and a directory of the same name. If your bucket's directory structure would result in this state, only the directory will be accessible. For example, our bucket contains the following object keys:
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:35]: Is this true? What happens with S3 Express One Zone?

**Review by dannycjones — COMMENTED** [review]
Thanks Ankit! I put some comments, sorry I didn't entirely review the troubleshooting document but there is some high-level feedback to action.

File: doc/TROUBLESHOOTING.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Accessing Glacier Objects
+Objects in the Glacier Flexible Retrieval and Glacier Deep Archive storage classes, and the Archive Access and Deep Archive Access tiers of S3 Intelligent-Tiering, are not accessible with Mountpoint. To access these objects with Mountpoint, restore or copy them to another storage class first.
+When listing or trying to access objects in these storage classes, the logs will show entries like:
+```
+WARN readdirplus{req=10 ino=1 fh=1 offset=0}: mountpoint_s3::inode: objects in \
+the GLACIER and DEEP_ARCHIVE storage classes are only accessible if restored
+```
+
+## Modifying file modification time and Sizes
```
Comment (monthonk) [inline doc/TROUBLESHOOTING.md:62]: We don't allow modifying metadata in general, so maybe
```suggestion
## Modifying metadata
```

File: doc/TROUBLESHOOTING.md (hunk 3 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:23]: echo gives different error -
`WARN setattr{req=12 ino=2 name="file.txt"}: mountpoint_s3::fuse: setattr failed: inode error: inode 2 (full key "file.txt") is a remote inode and its attributes cannot be modified`

Comment (sauraank) [reply doc/TROUBLESHOOTING.md:23]: I was trying to use vim. But, it is able to open the file using vim. I am just not able to overwrite. Not sure if I should add that as an example.

File: doc/TROUBLESHOOTING.md (hunk 6 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:35]: Yes. It is true for normal bucket. But, for S3 Express One zone, we are able to see both the file and directory. Also, this warning is not emitted. 
But, still reading the file is not possible. It gives error - 
`cat: out: Is a directory`

File: doc/TROUBLESHOOTING.md (hunk 3 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md]: I have updated the error according to my new PR #718 for the case of overwrite and modifying metadata.

File: doc/TROUBLESHOOTING.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: mountpoint_s3::fuse: write failed:\
+upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
+
+## Writing to an existing File 
+Mountpoint does support overwriting to existing file using mount option `--allow-overwrite` . Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+In the logs, we get the following WARN message - 
+```
+WARN file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
```
[the remark this answers is not in the document]
Comment (sauraank) [reply doc/TROUBLESHOOTING.md]: I am not sure what example to give for the case of opening file not in truncate mode. Adding a code snippet to do that will differ from other cases in this document. Let me dive deeper if there is any command line method to do it.

File: doc/TROUBLESHOOTING.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+Log entries for file overwriting looks like:
+
+```
+WARN setattr{req=4 ino=21 name="existing-file.txt"}: 
+mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified
+```
+
+### Unreleased
```
Comment (vladem) [inline doc/TROUBLESHOOTING.md:45]: this is hard to follow, may be put here only the final version and merge this page when the new release was created? probably, leaving a note like the following for prev versions, though:

> Note: in versions of MP before 1.5.0 the error will look like `mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified`

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 10]
```diff
@@ -48,6 +48,10 @@ body:
       render: shell
     validations:
       required: true      
+  - type: markdown
```
Comment (vladem) [inline .github/ISSUE_TEMPLATE/bug-report.yml:51]: may be put this to the top of the issue template? (e.g. right below the note about security issues) it would be a little frustrating if you already added some data (version, environment description) and than found the issue in the troubleshooting page

File: doc/TROUBLESHOOTING.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
+# Troubleshooting
```
Comment (vladem) [inline doc/TROUBLESHOOTING.md:1]: just as a possible source of inspiration for this page, I really liked how EKS structured their doc:
https://docs.aws.amazon.com/eks/latest/userguide/troubleshooting.html

basically, the problem/solution sections is a good template for all of the issues; not sure though, that all of the problems described in our doc have a solution/workaround, but it would be nice to think in that direction

Comment (sauraank) [reply doc/TROUBLESHOOTING.md:1]: In our doc, we are only adding logs for unsupported operations. So, they do not have any solutions. It is for users to know that it is expected behaviour.
Although, for some operations we can add solution like `allow-overwrite` and `allow-delete`.

File: doc/TROUBLESHOOTING.md (hunk 9 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:45]: I think we can merge this troubleshooting page for now and make the change once the new release is made. Because we don't know when are we going to do next patch release. If it is soon enough, I think we can wait.

File: doc/TROUBLESHOOTING.md [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+
+Mountpoint for Amazon S3 is optimized for applications that need high read throughput to large objects, potentially from many clients at once,
+and to write new objects sequentially from a single client at a time.
+To achieve this, Mountpoint does not implement all the features of a POSIX file system that may affect compatibility with your application.
+
+This documentation enumerates some examples of what error messages may look like when trying to perform unsupported operations. Please also take a look at Mountpoint's [semantics documentation](../doc/SEMANTICS.md) for more more detailed information on the decisions and tradeoffs made.
+This troubleshooting page is based on release Mountpoint-s3 v1.4.0 .Versions earlier to that might not have same error logging.
```
Comment (monthonk) [inline doc/TROUBLESHOOTING.md:8]:
```suggestion
This troubleshooting page is based on Mountpoint for Amazon S3 v1.4.0 or newer. Versions earlier to that might not have same error logging.
```

File: doc/TROUBLESHOOTING.md [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+## Modifying metadata
+
+Mountpoint does not support modifying metadata such as file modification times and file size. If user try to modify, for example file modification time using `touch`, user will get following error:
+ 
+```
+$ touch -a -m -t 201512180130.09 init.txt
+touch: init.txt: Operation not permitted
+```
+
+And the Mountpoint Logs will show:
```
Comment (monthonk) [inline doc/TROUBLESHOOTING.md:141]: probably
```suggestion
And Mountpoint logs will show:
```

File: doc/TROUBLESHOOTING.md [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+```
+WARN lookup{req=6 ino=1 name="class_GLACIER"}: 
+mountpoint_s3::inode: objects in the GLACIER and DEEP_ARCHIVE storage classes are only accessible if restored
+```
+
+## Modifying metadata
+
+Mountpoint does not support modifying metadata such as file modification times and file size. If user try to modify, for example file modification time using `touch`, user will get following error:
```
Comment (monthonk) [inline doc/TROUBLESHOOTING.md:134]:
```suggestion
Mountpoint does not support modifying metadata such as file modification times and file size. If users try to modify them, for example, updating file modification time with `touch`, user will get following error:
```

File: doc/TROUBLESHOOTING.md [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+In the logs, we get the following WARN message:
+
+```
+WARN setattr{req=11 ino=2 name="existing-file.txt"}: 
+mountpoint_s3::fuse: setattr failed: file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
+
+## Deleting file
+
+In order to delete files using Mountpoint user need to enable it using `--allow-delete` CLI flag. To know more about the behaviour of file deletion, please visit [Delete Semantics](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes)
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md]: As much as I hate to replace British English with American English... :)

```suggestion
In order to delete files using Mountpoint, you must opt-in using the `--allow-delete` CLI flag.
To learn more about how file deletion works in Mountpoint, please review the [file deletion section of Mountpoint's semantics documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#deletes).
```

File: doc/TROUBLESHOOTING.md [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+With improvement in error logging, we will get the following logs for file overwrite-
+
+In the logs, we get the following WARN message:
+
+```
+WARN setattr{req=11 ino=2 name="existing-file.txt"}: 
+mountpoint_s3::fuse: setattr failed: file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
+```
+
+## Deleting file
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:57]:
```suggestion
## Deleting files
```

File: doc/TROUBLESHOOTING.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+```
+WARN unlink{req=8 parent=1 name="test-file.txt"}: 
+mountpoint_s3::fuse: unlink failed: Deletes are disabled. Use '--allow-delete' mount option to enable it.
+```
+
+## Listing file which is shadowed by directory of same name
+
+A file system directory cannot contain both a file and a directory of the same name. If your bucket's directory structure would result in this state, only the directory will be accessible. The object will not be accessible. 
+For more details on how Mountpoint maps S3 object keys to files and directories, see the [semantics documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#mapping-s3-object-keys-to-files-and-directories). 
+For example, our bucket contains the following object keys:
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:78]: new line needed

```suggestion

For example, if a bucket contains the following object keys:
```

File: doc/TROUBLESHOOTING.md [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+
+* out
+* out/image.jpg
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:81]: Unless it makes sense (like `out` file), let's use things like `a`/`b` rather than `out`.

File: doc/TROUBLESHOOTING.md [hunk 19]
```diff
@@ -0,0 +1,143 @@
...
+$ ls
+out
+```
+
+Mountpoint logs will show an entry like this:
+
+```
+WARN readdir{req=5 ino=1 fh=2 offset=17}: 
+mountpoint_s3::inode::readdir::ordered:file 'out' (full key "out") is omitted because another directory 'out' exist with the same name.
+```
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:95]: malformed?

````suggestion
```
WARN readdir{req=5 ino=1 fh=2 offset=17}: 
mountpoint_s3::inode::readdir::ordered: file 'out' (full key "out") is omitted because another directory 'out' exist with the same name.
```
````

File: doc/TROUBLESHOOTING.md (hunk 9 above)
Comment (monthonk) [reply doc/TROUBLESHOOTING.md:46]: It might be helping users who build from source but also confusing for users that use official releases. I would not include it in the doc for now.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 20]
```diff
@@ -5,12 +5,16 @@ body:
   - type: markdown
     attributes:
       value: |
-        Thanks for taking the time to fill out a bug report.
+        Thanks for taking the time to fill out a bug report.
```
Comment (monthonk) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: clean the extra space.

File: doc/TROUBLESHOOTING.md [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+
+Although, in S3 express one zone bucket both file and directory will be shown without the above warning being emitted.
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:97]: That's a bug? If its a bug, we should open an issue and reference it.

We should also capitalize the name "S3 Express One Zone".

File: doc/TROUBLESHOOTING.md [hunk 22]
```diff
@@ -0,0 +1,143 @@
...
+```
+$ mv hello.txt hello_new.txt
+mv: rename hello.txt to hello_new.txt: Function not implemented
+```
+
+Mountpoint logs will show following message for the above error:
+
+```
+rename{req=2 parent=1 name="" newparent=1 newname=""}: mountpoint_s3::fuse: rename failed: operation not supported by Mountpoint
+```
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:115]: Is this example right? `name` and `newname` will really be blank?

File: doc/TROUBLESHOOTING.md [hunk 23]
```diff
@@ -0,0 +1,143 @@
...
+
+## Accessing Glacier objects
+
+Objects in the Glacier Flexible Retrieval and Glacier Deep Archive storage classes, and the Archive Access and Deep Archive Access tiers of S3 Intelligent-Tiering, are not accessible with Mountpoint. To access these objects with Mountpoint, restore or copy them to another storage class first.
+When listing or trying to access objects in these storage classes, Mountpoint logs will show entries like:
+
+```
+WARN readdirplus{req=10 ino=1 fh=1 offset=0}: 
+mountpoint_s3::inode: objects in the GLACIER and DEEP_ARCHIVE storage classes are only accessible if restored
+```
```
Comment (dannycjones) [inline doc/TROUBLESHOOTING.md:125]: Why do we warn on `readdirplus`? Should we fix that?

**Review by dannycjones — CHANGES_REQUESTED** [review]
I put a few comments nitpicking at how we present this information - sorry! It's really important we get this document clear and concise since customers will be reading it to understand problems they are facing when using Mountpoint.

File: doc/TROUBLESHOOTING.md (hunk 11 above)
Comment (dannycjones) [reply doc/TROUBLESHOOTING.md:1]: While we're starting with unsupported operations, I do like the EKS doc. I think you already have a structure very similar, Ankit, so we're on the right track!

File: doc/TROUBLESHOOTING.md [hunk 24]
```diff
@@ -0,0 +1,146 @@
...
+
+Mountpoint logs will show following message for the above error:
+
+```
+rename{req=2 parent=1 name="" newparent=1 newname=""}: mountpoint_s3::fuse: rename failed: operation not supported by Mountpoint
+```
+
+## Accessing Glacier objects
+
+Objects in the Glacier Flexible Retrieval and Glacier Deep Archive storage classes, and the Archive Access and Deep Archive Access tiers of S3 Intelligent-Tiering, are not accessible with Mountpoint. To access these objects with Mountpoint, restore or copy them to another storage class first.
```
Comment (monthonk) [inline doc/TROUBLESHOOTING.md:119]: maybe
```suggestion
Objects in Glacier Flexible Retrieval storage class, Glacier Deep Archive storage class, and non-instant access tiers of S3 Intelligent-Tiering storage class are not accessible with Mountpoint. To access these objects with Mountpoint, restore or copy them to another storage class first. See more details in [aws documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/archived-objects.html).
```

**Review by monthonk — COMMENTED** [review]
The format looks much nicer! we probably want to make wording a bit more clear.

File: doc/TROUBLESHOOTING.md (hunk 22 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:115]: No, it contains both the names in Linux. I was checking all the error messages from Ubuntu 22.04, but somehow I checked this from MacOS. I should update it.

File: doc/TROUBLESHOOTING.md (hunk 23 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:125]: This error is propagated from `is_readable()` [method](https://github.com/awslabs/mountpoint-s3/blob/92aec0a0132aaf8d20093943cb40269bb57ffb2f/mountpoint-s3/src/inode.rs#L1488-L1503) implemented on InodeStat.

Comment (dannycjones) [reply doc/TROUBLESHOOTING.md:125]: Ah I see... I'd probably just include one of these two logs since they are very similar.

File: doc/TROUBLESHOOTING.md (hunk 9 above)
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:46]: In that case, I think we can keep troubleshooting page updated with #718 and once that PR gets released in something like v1.4.1, we can merge this PR (soon after or before??)
Also, how do we make sure that this PR is ready to be merged but not merge it till next release? Just approve it and wait? We might need to rebase (although unlikely) and be careful to not merge it before the release.

File: doc/TROUBLESHOOTING.md [hunk 25]
```diff
@@ -0,0 +1,143 @@
...
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+
+In Mountpoint's logs, a warning message will be emitted similar to below:
+
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: 
+mountpoint_s3::fuse: write failed:upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
```
[the remark this answers is not in the document]
Comment (sauraank) [reply doc/TROUBLESHOOTING.md:24]: Yes. Earlier I was breaking the line at that point. In modifying that, it got malformed.

File: doc/TROUBLESHOOTING.md (hunk 9 above)
Comment (vladem) [reply doc/TROUBLESHOOTING.md:46]: Yeah, its frustrating to have PRs open for too long. You could put something like this to unblock the merge:
___

Log entries for file overwrites looks like one of the following depending on the Mountpoint version:
```
WARN setattr{req=4 ino=21 name="existing-file.txt"}: 
mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified
```
```
WARN setattr{req=11 ino=2 name="existing-file.txt"}: 
mountpoint_s3::fuse: setattr failed: file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
```
___

It's just this `Unreleased` looks a little off here.

File: doc/TROUBLESHOOTING.md (hunk 24 above)
Comment (dannycjones) [reply doc/TROUBLESHOOTING.md:119]: nice! just capitalize AWS

**Comment by sauraank** [comment]
For the overwrite error logging, I am keeping troubleshooting page as the error is logged in latest release. Changes in its `setattr` error logging which is merged in #718 can be reflected once we make the next release. It will just changing one error message. I do not want to block this PR till next patch release to get merged.
Also, we can have tag for documentation changes. What do you think @monthonk @dannycjones ?

File: README.md [hunk 26]
```diff
@@ -14,8 +14,12 @@ but probably not the right fit for applications that:
 * use file operations that S3 doesn't natively support, like directory renaming or symlinks
 * make edits to existing files (don't work on your Git repository or run `vim` in Mountpoint 😄)
 
+
```
Comment (monthonk) [inline README.md:17]: nit: remove this line

File: doc/TROUBLESHOOTING.md [hunk 27]
```diff
@@ -0,0 +1,145 @@
...
+
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: 
+mountpoint_s3::fuse: write failed: upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
+
+## Writing to an existing file 
+
+Trying to open an existing file for writing using Mountpoint without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+For example: for a directory with existing file 'existing-file.txt', the following error will occur when trying to overwrite it:
```
Comment (monthonk) [inline doc/TROUBLESHOOTING.md:32]:
```suggestion
For example, overwriting an existing file in Mountpoint would result in the following error:
```

**Review by monthonk — DISMISSED** [review]
lgtm, thanks Ankit!

**Review by dannycjones — DISMISSED** [review]
LGTM. I pushed a few minor changes, I think this is ready to merge!

**Review by dannycjones — APPROVED** [review]
Oops, my bad. Now ready to merge. :)
